src/python/grpcio/grpc/experimental/aio/_interceptor.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2019 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Interceptors implementation of gRPC Asyncio Python."""
+import collections
+from typing import Any, Dict, Callable, Optional
+
+import grpc
+
+
+class ClientCallDetails(
+        collections.namedtuple(
+            'ClientCallDetails',
+            ('method', 'timeout', 'metadata', 'credentials')),
+        grpc.ClientCallDetails):
+
+    method: int
+    timeout: Optional[float]
+    metadata: Optional[Dict]
+    credentials: Optional[Any]
+
+
+class UnaryUnaryClientInterceptor:
+    """Affords intercepting unary-unary invocations."""
+
+    async def intercept_unary_unary(
+            self, continuation: Callable[[ClientCallDetails, Any], Any],
```

Comment:
umm, I would say no since the continuation is still a `Callable` object and with the current version, we are returning the response which is not an _awaitable_ object.

If finally the interceptors are changed for supporting the `Call` object we would need to change this for `Callable[[ClientCallDetails, Any], Call]`, but this is still under discussion.

Let's first address this discussion.